enhancements/olm-cmd-workflows.md

```diff
@@ -0,0 +1,164 @@
+
+## Goals
+
+The proposal is aimed at explaining the resulting process of developing an OLM enabled operator from scratch with the above mentioned improvements.
+
+## Proposal
+
+### README
+
+The following document discusses workflows for creating a new operator and running it with OLM:
+
+Operator creation workflow:
+
+1. Create a new operator project using the SDK Command Line Interface(CLI)
+2. Define new resource APIs by adding Custom Resource Definitions(CRD)
+3. Define Controllers to watch and reconcile resources
+4. Write the reconciling logic for your Controller using the SDK and controller-runtime APIs
+5. Use the SDK CLI to build and generate the operator deployment manifests
+
+OLM Workflow:
+
+1. Use `operator-sdk generate bundle` to generate kustomize base templates with CSV metadata, build and push the operator bundle image to the specified registry.
+  * If the kustomize template having the operator and UI metadata is present, it is utilized along with the existing CRDs to build the operator image bundle.
+  * If the kustomize template is not present, provide inputs regarding the UI metadata to the interactive prompts appearing further.
+2. Use `operator-sdk run --olm` to generate kustomize base template with operator metadata, build and run the operator.
+
+Command details:
+
+1. `operator-sdk generate bundle`:
+
+The `operator-sdk generate bundle` command will create a kustomize base template containing metadata for the generation of CSV, build and push the operator image bundle to the specified registry.
+
+```
+Usgae:
+    operator-sdk generate bundle <IMG_REGISTRY>/<TAG> [optional flags]
+
+    Flags:
+        -p, --package string    (optional) The name of the package that bundle belongs to. Set if package name differs from project name.
+        -d, --kustomize-dir     (optional) Path to the directory containing kustomize base templates. Default: ./config.
+        -o, --output-dir        (optional) Optional output directory if operator manifests are to be written on disk.
+
+If the command is run on a vanilla Operator-SDK project or when the existing kustomize template does not contain UI metadata for generating CSV, the command will be followed by a series of prompts asking the user for inputs. Example:
+$ Provide DisplayName for your operator:		
+$ Provide version for your operator:		
+$ Provide minKubeVersion for CSV:
+$ Provide the list of maintainers for the operator:
+$ Provide the list of relevant links for the operator:
+```
+
+2. `operator-sdk run --olm`:
+
+The `operator-sdk run --olm` command will generate kustomize base templates conataining operator metadata and deploy the operator on cluster. It can be used to rapidly build, deploy and test your operator using OLM. The kustomize templates to be used for generating manifests can also be specified using the `--kustomize-dir` flag.
```

Comment:
```suggestion
The `operator-sdk run --olm` command will render Kubernetes resources from the kustomize base templates conataining operator metadata and deploy the operator on cluster. It can be used to rapidly build, deploy and test your operator using OLM. The kustomize templates to be used for generating manifests can also be specified using the `--kustomize-dir` flag.
```